text/0152-telemetry-schemas.md

```diff
@@ -0,0 +1,1159 @@
+  Future Possibilities section).
+
+- We intentionally limit the types of transformations of schemas to the bare
+  minimum that is necessary to handle the most common changes that we believe
+  OpenTelemetry Semantic Conventions will require in the near future. More types
+  of transformations [may be proposed](#other-transformation-types) in the
+  future. This proposal does not attempt to support a comprehensive set of
+  possible transformation types that can handle all possible changes to schemas
+  that we can imagine. That would be too complicated and very likely
+  superfluous. Any new transformation types should be proposed and added in the
+  future to the schema file format when there is an evidence that they are
+  necessary for the evolution of OpenTelemetry.
+
+## Use Cases
+
+This section shows a couple interesting use-cases for the telemetry schemas
+(other uses-cases are also possible, this is not an exhaustive list).
+
+### Full Schema-Aware
+
+Here is an example on a schema-aware observability system:
+
+![Full Schema-Aware](img/0152-otel-schema.png)
+
+Let's have a closer look at what happens with the Telemetry Source and Backend
+pair as the telemetry data is emitted, delivered and stored:
+
+![Source and Backend](img/0152-source-and-backend.png)
+
+In this example the telemetry source produces spans that comply with version
+1.2.0 of OpenTelemetry schema, where "deployment.environment" attribute is used
+to record that the span is coming from production.
+
+The telemetry consumer desires to store the telemetry in version 1.1.0 of
+OpenTelemetry schema. The schema translator compares the schema_url in the
+received span with the desired schema and sees that a version conversion is
+needed. It then applies the change that is described in the schema file and
+renames the attribute from "deployment.environment" to "environment" before
+storing the span.
+
+And here is for example how the schemas can be used to query stored data:
+
+![Query Translate](img/0152-query-translate.png)
+
+### Collector-Assisted Schema Transformation
+
+Here is a somewhat different use case, where the backend is not aware of schemas
+and we rely on OpenTelemetry Collector to translate the telemetry to a schema
+that the backend expects to receive. The "Schema Translate Processor" is
+configured, the target schema_url is specified and all telemetry data that
+passes through the Collector is converted to that target schema:
+
+![Collector](img/0152-collector.png)
+
+## Schema URL
+
+Schema URL is an identifier of a Schema. The URL specifies a location of a
+[Schema File](#schema-file) that can be retrieved (so it is a URL and not just a
+URI) using HTTP or HTTPS protocol.
+
+Fetching the specified URL may return an HTTP redirect status code. The fetcher
+MUST follow the HTTP standard and honour the redirect response and fetch the
+file from the redirected URL.
+
+The last part of the URL path is the version number of the schema.
```

Comment:
Per my comment above, I think this should be in the list of requirements :)

Reply:
I did not want to overload the Summary section with too many details. Let me see if I can add this.

Reply:
Reworded the corresponding item in Summary section.